### PR TITLE
feat: getIdByNumber function added to check the real phone number

### PR DESCRIPTION
**Pull request by felipeDS91**
Here in Brazil we have a problem with some phone numbers that doesn't have the number "9" in front the phone number, because some years ago, all phone numbers of the country received an additional number "9". So when you try to send a message for one of these old numbers with the new pattern (with "9"), whatsapp thinks that's a different number and starts a new chat.
With this function, we can send the number always with the "9" digit and if it is an old number, this function will return the right number (without "9").

### CODE REVIEW

File: src/Client.js [hunk 1]
```diff
@@ -712,6 +712,18 @@ class Client extends EventEmitter {
...
 
+    /**
+     * Returns a real id registered in whatsapp
+     * @param {string} id the whatsapp user's ID
+     * @returns {Promise<string>}
+     */
+    async getIdByNumber(id) {
+        return await this.pupPage.evaluate(async (id) => {
+            let result = await window.Store.Wap.queryExist(id);
+            return result.status === 200 ? result.jid._serialized : undefined;
```
Comment (vitorleonel) [inline src/Client.js:723]: Hey,

Do you think it could be better if it returned null when you didn't find it?

Comment (felipeDS91) [reply src/Client.js:723]: Yes, of sure. I will change it.

Comment (felipeDS91) [reply src/Client.js]: Thanks by suggestion.

**Review by vitorleonel — APPROVED** [review]

**Comment by lucasrosa90** [comment]
Dont need this PR, the lib already support that.

```javascript
try {
  const conversation = await this.client.pupPage.evaluate(
    async (number) => {
      return await window.WWebJS.getNumberId(number);
    },
    id
  );
  console.log(`Registered chat ID: ${conversation._serialized}`);
      return conversationId._serialized;
} catch (err) {
  console.error(`Chat ID not registered.`);
}
```

**Comment by pedroslopez** [comment]
The `getNumberById` function was being used in the past for sending messages to new contacts, but the implementation has changed and it's not being used by the library for anything right now.

If this is the case and it helps with brazilian numbers, maybe instead it would be good to expose the function directly in the client through something like `client.getNumberId()`.

**Comment by pedroslopez** [comment]
Closing in favor of #483

**Pull request closed without merging**